### DISCUSSION TITLE
Install extension from GitHub private repo

### DISCUSSION THREAD

**Discussion by abichat** [quarto-dev/quarto-cli#4444] [Feature Requests]
Hello!

Installing a extension from a private repo doesn't work, even if credentials are stored:
```
quarto add privateorg/ext
Extension not found in local or remote sources
```

Would it be possible to use the stored credentials, or ask them? 

Meanwhile, a workaround could be done like this:

```
git clone https://github.com/privateorg/ext path/to/the/local/repo/ext
quarto add path/to/the/local/repo/ext
```

Thanks!

**Comment by cderv** [comment]
@dragonstyle I wonder if we could bring support for api download which allows credentials or even `gh` cli if available that also allow auth. This would allow quarto to rely on external credentials configuration to access private repo. 

What do you think ?

**Comment by mcanouil** [comment]
Related to #5827

**Comment by jan-abel-inwt** [comment]
Hello!

I appreciate the work done on quarto. I'm writing to inquire if there are any plans to include this feature in upcoming updates?

Thanks!

**Comment by mcanouil** [comment]
I am going to close this in favour of #5827.

@abichat Watch #5827 (which I reopened since it was wrongly closed).
